enhancements/automated-resource-request-scaling.md

```diff
@@ -0,0 +1,635 @@
+   could be changed in later versions without updating the source for
+   any operators. In order to change the behavior in an upgraded
+   cluster, something would have to modify the Infrastructure config
+   resource in the existing cluster.
+
+2. Expose the value as a `const` in the library where the
+   Infrastructure config API is defined.
+
+   We could publish the value only in `go` source code, using a
+   `const` in the same module where the Infrastructure API `struct` is
+   defined. Consumers of the API that import the code would be able to
+   use the value, and receive updates by updating to a new version of
+   the library.
+
+   If the value changes between releases, upgraded operators would
+   have the new value.
+
+### Dynamic vs. Static Setting
+
+It is not likely that users would need to change the setting after
+building a cluster, so we could have the installer use a value from
+the `install-config.yaml` to set a status field in whatever CRD we
+choose for the API. If we do need to support scaling later, that work
+can be described in another enhancement.
+
+### Managing settings for operators installed by cluster-version-operator
+
+The alternatives in this subsection cover the need to change the
+settings for the static manifests currently managed by the
+cluster-version-operator. These approaches would only apply to the
+operators themselves, and not their operands.
+
+1. Use a cluster profile to change the resource requests for control
+   plane operators
+
+   We could create a separate cluster profile with different resource
+   requests for our control plane components.
+
+   This would be quite a lot of work to implement and test.
+
+   We may eventually need different profiles for different customers,
+   further multiplying the amount of work.
+
+   As an organization, we are trying to keep the number of profiles
+   small as a general rule.
+
+   Profile settings only apply to the cluster operators Deployments,
+   and changing the resource requests in the static manifests does not
+   signal to the operators that they need to change the settings for
+   their operands.
+
+2. Lower the resource requests for cluster operators statically in the
+   default configuration
+
+   We could apply changes to the default resource requests in the
+   static manifests used by all deployments, instead of adding one or
+   more new cluster profiles.
+
+   This would affect all clusters, and if we are too aggressive with
+   the changes we may lower values too much and cause important
+   cluster operators to be starved for resources, triggering degraded
+   cluster performance.
+
+   The static settings only apply to the cluster operators
+   Deployments, and changing them does not signal to the operators
+   that they need to change the settings for their operands.
+
+   With lowered CPU requests, the consumption of some operators may
+   appear more sporadic ("bursty") than it does currently. One way to
```

Comment:
Not sure I understand this statement. The absolute cpu usage of any spikes should not change.

Reply:
I phrased this poorly. What I was trying to capture was @wking's comments about restructuring some operators so that they live within their requested resources (so those requests continue to look accurate), and the periodic "bursts" of work are run in a way that means they can be scheduled separately.